<?xml version="1.0"?>

<launch>
    <arg name="map_file" default="$(find chibi20_2)/mapdata/goodmap.yaml"/>
    <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)" output="screen"/>

    <!-- <include file="$(find chibi21_b)/launch/Localizer_path.launch" /> -->

    <!-- rviz : rvizz.launch -->

    <!-- rosbag play -->
    <param name="use_sim_time" value="true" type="bool"/>
    <arg name="roomba_num" default="0" />
    <group if="$(eval roomba_num==1)">
        <node pkg="rosbag" type="play" name="play"
            args="--clock /home/amsl/bagfiles/with_yolo2/test.bag -r 2"
            required="true">
        </node>
    </group>
    <group if="$(eval roomba_num==2)">
        <node pkg="rosbag" type="play" name="play"
            args="--clock /home/amsl/bagfiles/with_yolo2/roomba4.bag
                          /home/amsl/bagfiles/with_yolo2/roomba3.bag"
            required="true" >
        </node>
    </group>
    <group if="$(eval roomba_num==3)">
        <node pkg="rosbag" type="play" name="play"
            args="--clock /home/amsl/bagfiles/with_yolo2/roomba1.bag
                          /home/amsl/bagfiles/with_yolo2/roomba2.bag
                          /home/amsl/bagfiles/with_yolo2/roomba3.bag"
            required="true" >
        </node>

    </group>
    <group if="$(eval roomba_num==4)">
        <node pkg="rosbag" type="play" name="play"
            args="--clock /home/amsl/bagfiles/with_yolo2/roomba1.bag
                          /home/amsl/bagfiles/with_yolo2/roomba2.bag
                          /home/amsl/bagfiles/with_yolo2/roomba3.bag
                          /home/amsl/bagfiles/with_yolo2/roomba4.bag"
            required="true" >
        </node>
    </group>
    <group if="$(eval roomba_num==5)">
        <node pkg="rosbag" type="play" name="play"
            args="--clock /home/amsl/bagfiles/with_yolo2/roomba1.bag
                          /home/amsl/bagfiles/with_yolo2/roomba2.bag
                          /home/amsl/bagfiles/with_yolo2/roomba3.bag
                          /home/amsl/bagfiles/with_yolo2/roomba4.bag
                          /home/amsl/bagfiles/with_yolo2/roomba5.bag"
            required="true" >
        </node>
    </group>
    <group if="$(eval roomba_num==6)">
        <node pkg="rosbag" type="play" name="play"
            args="--clock /home/amsl/bagfiles/with_yolo2/roomba1.bag
                          /home/amsl/bagfiles/with_yolo2/roomba2.bag
                          /home/amsl/bagfiles/with_yolo2/roomba3.bag
                          /home/amsl/bagfiles/with_yolo2/roomba4.bag
                          /home/amsl/bagfiles/with_yolo2/roomba5.bag
                          /home/amsl/bagfiles/with_yolo2/roomba6.bag"
            required="true" >
        </node>
    </group>

    <!-- localizer node -->
    <arg name="output" default="log"/>
    <!-- roomba1 -->
    <node pkg="distance_based_localizer" type="distance_based_localizer_ver3" name="distance_based_localizer_1" output="$(arg output)" >
        <remap from="/roomba/odometry" to="roomba1/roomba/odometry" />
        <remap from="/object_positions" to="roomba1/object_positions" />
        <remap from="/db_pose" to="roomba1/db_pose" />
        <remap from="/db_poses" to="roomba1/db_poses" />
        <remap from="/score" to="roomba1/score" />
        <remap from="/path" to="roomba1/path" />
        <remap from="/behind_roomba/score" to="/roomba2/score" />
        <rosparam command="load" file="$(find distance_based_localizer)/config/roomba1_ver3.yaml" />
        <param name="behind_roomba_th" value="1.0" type="double" />
        <param name="behind_th" value="0.2" type="double" />
    </node>

    <!-- roomba2 -->
    <node pkg="distance_based_localizer" type="distance_based_localizer_ver3" name="distance_based_localizer_2" output="$(arg output)" >
        <remap from="/roomba/odometry" to="roomba2/roomba/odometry" />
        <remap from="/object_positions" to="roomba2/object_positions" />
        <remap from="/db_pose" to="roomba2/db_pose" />
        <remap from="/db_poses" to="roomba2/db_poses" />
        <remap from="/score" to="roomba2/score" />
        <remap from="/path" to="roomba2/path" />
        <remap from="/front_roomba/score" to="roomba1/score" />
        <remap from="/behind_roomba/score" to="roomba3/score" />
        <rosparam command="load" file="$(find distance_based_localizer)/config/roomba2_ver3.yaml" />
        <param name="behind_th" value="0.2" type="double" />
        <param name="behind_roomba_th" value="1.0" type="double" />
        <param name="front_roomba_th" value="0.8" type="double" />
    </node>

    <!-- roomba3 -->
    <node pkg="distance_based_localizer" type="distance_based_localizer_ver3" name="distance_based_localizer_3" output="$(arg output)" >
        <remap from="/roomba/odometry" to="roomba3/roomba/odometry" />
        <remap from="/object_positions" to="roomba3/object_positions" />
        <remap from="/db_pose" to="roomba3/db_pose" />
        <remap from="/db_poses" to="roomba3/db_poses" />
        <remap from="/score" to="roomba3/score" />
        <remap from="/path" to="roomba3/path" />
        <remap from="/front_roomba/score" to="roomba2/score" />
        <remap from="/behind_roomba/score" to="roomba4/score" />
        <rosparam command="load" file="$(find distance_based_localizer)/config/roomba3_ver3.yaml" />
        <param name="behind_th" value="0.2" type="double" />
        <param name="behind_roomba_th" value="1.1" type="double" />
        <param name="front_roomba_th" value="0.8" type="double" />
    </node>

    <!-- roomba4 -->
    <node pkg="distance_based_localizer" type="distance_based_localizer_ver3" name="distance_based_localizer_4" output="$(arg output)" >
        <remap from="/roomba/odometry" to="roomba4/roomba/odometry" />
        <remap from="/object_positions" to="roomba4/object_positions" />
        <remap from="/db_pose" to="roomba4/db_pose" />
        <remap from="/db_poses" to="roomba4/db_poses" />
        <remap from="/score" to="roomba4/score" />
        <remap from="/path" to="roomba4/path" />
        <remap from="/front_roomba/score" to="roomba3/score" />
        <remap from="/behind_roomba/score" to="roomba5/score" />
        <rosparam command="load" file="$(find distance_based_localizer)/config/roomba4_ver3.yaml" />
        <param name="behind_th" value="0.2" type="double" />
        <param name="behind_roomba_th" value="1.0" type="double" />
        <param name="front_roomba_th" value="0.8" type="double" />
    </node>

    <!-- roomba5 -->
    <node pkg="distance_based_localizer" type="distance_based_localizer_ver3" name="distance_based_localizer_5" output="$(arg output)" >
        <remap from="/roomba/odometry" to="roomba5/roomba/odometry" />
        <remap from="/object_positions" to="roomba5/object_positions" />
        <remap from="/db_pose" to="roomba5/db_pose" />
        <remap from="/db_poses" to="roomba5/db_poses" />
        <remap from="/score" to="roomba5/score" />
        <remap from="/path" to="roomba5/path" />
        <remap from="/front_roomba/score" to="roomba4/score" />
        <remap from="/behind_roomba/score" to="roomba6/score" />
        <rosparam command="load" file="$(find distance_based_localizer)/config/roomba5_ver3.yaml" />
        <param name="behind_th" value="0.2" type="double" />
        <param name="behind_roomba_th" value="1.0" type="double" />
        <param name="front_roomba_th" value="0.8" type="double" />
    </node>

    <!-- roomba6 -->
    <node pkg="distance_based_localizer" type="distance_based_localizer_ver3" name="distance_based_localizer_6" output="$(arg output)" >
        <remap from="/roomba/odometry" to="roomba6/roomba/odometry" />
        <remap from="/object_positions" to="roomba6/object_positions" />
        <remap from="/db_pose" to="roomba6/db_pose" />
        <remap from="/db_poses" to="roomba6/db_poses" />
        <remap from="/score" to="roomba6/score" />
        <remap from="/path" to="roomba6/path" />
        <remap from="/front_roomba/score" to="roomba5/score" />
        <rosparam command="load" file="$(find distance_based_localizer)/config/roomba6_ver3.yaml" />
        <param name="behind_th" value="0.2" type="double" />
        <param name="behind_roomba_th" value="1.0" type="double" />
        <param name="front_roomba_th" value="0.8" type="double" />
    </node>

    <!-- roomba_odom -->
    <arg name="add_odom" default="false" /> <!-- unless : nothing will happen -->
    <group if="$(arg add_odom)">
        <node pkg="distance_based_localizer" type="distance_based_localizer_odom" name="distance_based_localizer_odom" output="$(arg output)" >
                <remap from="/roomba/odometry" to="roomba3/roomba/odometry" />
                <remap from="/db_pose" to="roomba4_odom/db_pose" />
                <remap from="/db_poses" to="roomba4_odom/db_poses" />
                <remap from="/path" to="roomba4_odom/path" />

                <param name="roomba_name" value="3" type="int" />
                <param name="x_init" value="7.5" type="double" />
                <param name="y_init" value="0.0" type="double" />
            </node>
        </group>
    <!-- roomba4_solo -->
    <arg name="add_solo" default="false" /> <!-- unless : nothing will happen -->
    <group if="$(arg add_solo)">
        <node pkg="distance_based_localizer" type="distance_based_localizer_ver3" name="distance_based_localizer_solo" output="$(arg output)" >
                <remap from="/roomba/odometry" to="roomba3/roomba/odometry" />
                <remap from="/object_positions" to="roomba3/object_positions" />
                <remap from="/db_pose" to="roomba4_solo/db_pose" />
                <remap from="/db_poses" to="roomba4_solo/db_poses" />
                <remap from="/path" to="roomba4_solo/path" />

                <rosparam command="load" file="$(find distance_based_localizer)/config/roomba3_ver3.yaml" />
            </node>
        </group>

    <!-- <!&#45;&#45; evaluations &#45;&#45;> -->
    <!--     <arg name="bagfile_num" default=""/> -->
    <!-- <!&#45;&#45; recording : /eva &#38; /score &#45;&#45;> -->
    <!-- <include file="$(find distance_based_localizer)/launch/evaluation.launch"> -->
    <!--     <arg name="roomba" value="1"/> -->
    <!--     <arg name="rec_bag_name" value="roomba1$(arg bagfile_num)"/> -->
    <!-- </include> -->
    <!-- <include file="$(find distance_based_localizer)/launch/evaluation.launch"> -->
    <!--     <arg name="roomba" value="2"/> -->
    <!--     <arg name="rec_bag_name" value="roomba2$(arg bagfile_num)"/> -->
    <!-- </include> -->
    <!-- <include file="$(find distance_based_localizer)/launch/evaluation.launch"> -->
    <!--     <arg name="roomba" value="3"/> -->
    <!--     <arg name="rec_bag_name" value="roomba3$(arg bagfile_num)"/> -->
    <!-- </include> -->
    <!-- <include file="$(find distance_based_localizer)/launch/evaluation.launch"> -->
    <!--     <arg name="roomba" value="4"/> -->
    <!--     <arg name="rec_bag_name" value="roomba4$(arg bagfile_num)"/> -->
    <!-- </include> -->
    <!-- <include file="$(find distance_based_localizer)/launch/evaluation.launch"> -->
    <!--     <arg name="roomba" value="5"/> -->
    <!--     <arg name="rec_bag_name" value="roomba5$(arg bagfile_num)"/> -->
    <!-- </include> -->
    <!-- <include file="$(find distance_based_localizer)/launch/evaluation.launch"> -->
    <!--     <arg name="roomba" value="6"/> -->
    <!--     <arg name="rec_bag_name" value="roomba6$(arg bagfile_num)"/> -->
    <!-- </include> -->

</launch>

